lib/src/transaction.rs

```diff
@@ -202,12 +202,30 @@ impl UnpublishedOperation {
         let repo = self
             .repo_loader
             .create_from(data.operation, data.view, data.index);
-        self.closed = true;
+        self.closed.mark_closed();
         repo
     }
 }
 
-impl Drop for UnpublishedOperation {
+// Used to ensure `UnpublishedOperation`s are either published or explicitly
+// left unpublished before they're dropped. Using a separate type to do this
+// instead of implementing `Drop` for `UnpublishedOperation` allows fields to be
+// moved from `UnpublishedOperation` without being wrapped in `std::Option`.
+struct MustClose {
+    closed: bool,
+}
+
+impl MustClose {
+    fn new() -> MustClose {
+        MustClose { closed: false }
+    }
+
+    fn mark_closed(&mut self) {
+        self.closed = true;
+    }
+}
+
+impl Drop for MustClose {
```

Comment:
I don't know why we use `Drop` + `panic()` here, but I think `#[must_use]` on `UnpublishedOperation` is good enough to catch coding error. It's not a linear type, but it's probably okay if the user explicitly "use" the unpublished op.

Reply:
Yeah, I wasn't sure either, and I agree, `#[must_use]` is sufficient, and maybe even better since it will generate a compiler error in most cases.